<launch>
  <env name="ROSCONSOLE_CONFIG_FILE"
       value="$(find k4a_arm_support)/log/rosconsole.conf"/>

  <node pkg="cv_camera" type="cv_camera_node" name="cv_camera" output="screen">
    <param name="rate" type="double" value="15.0"/>
    <param name="camera_info_url" type="string" value="package://k4a_arm_support/calibration/lifecam_studio.yaml"/>
    <param name="frame_id" type="string" value="ros_msft_onnx_link"/>
  </node>
  <!--
  <node pkg="image_publisher" type="image_publisher" name="cv_camera"
        args="$(find k4a_arm_support)/launch/image.png" >
    <param name="flip_horizontal" value="false" />
    <param name="flip_vertical" value="false" />
    <param name="frame_id" value="ros_msft_onnx_link" />
    <param name="publish_rate" value="5" />
    <param name="camera_info_url" value="package://k4a_arm_support/calibration/lifecam_studio.yaml" />
  </node>
  -->

  <node pkg="tuw_aruco" type="aruco_node" name="arMarkerAruco" output="screen">
    <param name="show_debug_image" value="true"/>

    <param name="marker_dictonary" value="ARUCO_MIP_16h3"/>
    <param name="marker_size" value="0.03"/>

    <param name="publish_tf" value="true"/>
    <param name="publish_markers" value="true"/>
    <param name="publish_fiducials" value="false"/>
    <param name="pose_estimation_enabled" value="true"/>


    <remap from="image" to="/cv_camera/image_raw"/>
    <remap from="camera_info" to="/cv_camera/camera_info"/>

    <remap from="markers" to="/markersAruco"/>
  </node>
</launch>